js/boaexchange.js

```diff
@@ -0,0 +1,1163 @@
+        for (let i = 0; i < codes.length; i++) {
+            let code = codes[i];
+            let currency = this.currency (code);
+            let request = { 'label': currency['id'] };
+            let response = await this.v1GetCoinsLabel (request);
+            withdrawFees[code] = this.safeFloat (response['data'], 'tx_fee');
+        }
+        return {
+            'withdraw': withdrawFees,
+            'deposit': {},
+            'info': info,
+        };
+    }
+
+    async fetchFundingLimits (symbols = undefined, params = {}) {
+        // this method should not be called directly, use loadTradingLimits () instead
+        // by default it will try load withdrawal fees of all currencies (with separate requests, sequentially)
+        // however if you define symbols = [ 'ETH/BTC', 'LTC/BTC' ] in args it will only load those
+        await this.loadMarkets ();
+        if (symbols === undefined) {
+            symbols = this.symbols;
+        }
+        let result = {};
+        for (let i = 0; i < symbols.length; i++) {
+            let symbol = symbols[i];
+            result[symbol] = await this.fetchFundingLimitsById (this.marketId (symbol), params);
+        }
+        return result;
+    }
+
+    async fetchFundingLimitsById (id, params = {}) {
+        let request = {
+            'symbol': id,
+        };
+        let response = await this.v1GetCoinsLabelLimits (this.extend (request, params));
+        return this.parseFundingLimits (this.safeValue (response, 'data', {}));
+    }
+
+    async fetchLedger (code = undefined, since = 0, limit = 0, params = {}) {
+        await this.loadMarkets ();
+        let request = {
+            'page': since,
+            'limit': limit,
+        };
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+            request['coin'] = currency['id'];
+        }
+        const response = await this.v1GetLedger (this.extend (request, params));
+        return this.parseLedgerEntries (response['data'], currency);
+    }
+
+    async fetchMarkets (params = {}) {
+        const response = await this.v1GetMarkets ();
+        return this.parseMarkets (response['data']);
+    }
+
+    async fetchMyTrades (symbol = undefined, since = 0, limit = 0, params = {}) {
+        params['symbol'] = symbol;
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let response = await this.v1GetTrades (this.extend ({
+            'market': market['id'],
+            'begin': since,
+            'limit': limit,
+        }, params));
+        if ('data' in response) {
+            if (response['data'] !== undefined) {
+                // Re-format dat
+                let data = [];
+                for (let i = 0; i < response['data'].length; i++) {
+                    let order = response['data'][i];
+                    data.push ({
+                        'market': order['market'],
```

Comment:
This has to be moved to `parseTrade` and checked-for there, in other words, we want to keep the separation of concerns, so, all about format is parsing, therefore we'd rather move as much of it as possible to the parsing method, rather than keep it in the fetching method. I know there's other exceptions from this rule with other exchange files, but we're not proud of them )